connectors/byoc.py

```diff
@@ -159,10 +160,11 @@ def _create_object(self, doc_source):
 
 
 class SyncJob:
-    def __init__(self, connector_id, elastic_client):
+    def __init__(self, connector_id, elastic_client, filtering):
```

Comment:
This will be refactored. the current `start` function will be split to 
1. `create` function, which index a new doc in `.elastic-connectors-sync-jobs`
2. `claim` function, which change job status from `pending` to `in_progress`.

You can proceed without waiting for the refactor. But I think the `filtering` should be a param of `start` function, not `__init__`.

Reply:
Good to know, will move it `start`, makes more sense 👍

Reply:
Done